mesonbuild/modules/_qt.py

```diff
@@ -75,11 +77,13 @@ class PreprocessKwArgs(TypedDict):
         include_directories: T.List[T.Union[str, build.IncludeDirs]]
         dependencies: T.List[T.Union[Dependency, ExternalLibrary]]
         method: str
+        output_json: bool
         preserve_paths: bool
 
     class HasToolKwArgs(kwargs.ExtractRequired):
 
         method: str
+        tools: T.List[str]
```

Comment:
you can make this a little more powerful (and maybe remove an assert later on) with:
`tools: T.List[Literal['moc, 'lrelease', <insert reset of valid tools>]]`

Reply:
fixed